tests/platform/platform_api.cpp

```diff
@@ -26,6 +26,54 @@
 namespace platform_api__ {
 using namespace sycl_cts;
 
+// Compares two devices by their hash value.
+struct DeviceHashLessT {
+  bool operator()(const sycl::device& lDevice, const sycl::device& rDevice) {
+    std::hash<sycl::device> hasher;
+    return hasher(lDevice) < hasher(rDevice);
+  }
+};
+
+// Checks that all devices in a vector are unique.
+inline bool AllDevicesUnique(const std::vector<sycl::device> &devices) {
+  std::vector<sycl::device> devicesCopy = devices;
+  std::sort(devicesCopy.begin(), devicesCopy.end(), DeviceHashLessT{});
+  return std::unique(devicesCopy.begin(), devicesCopy.end()) ==
+         devicesCopy.end();
```

Comment:
Learning about std::unique. I don't know if it's better or ulgy than the old school `set{}.size` comparaison :p